### DISCUSSION TITLE
Recommended pattern for applying a polynomial of sdpvar variables to a large matrix

### DISCUSSION THREAD

**Discussion by htadashi** [yalmip/YALMIP#1249] [General discussion] [answered]
Hi @johanlofberg,

I need to apply a polynomial of sdpvar variables to a very large matrix. Unfortunately, I didn't come up with a solution (or a prompt to ChatGPT) to optimize this (probably bad) pattern:

```matlab
sdpvar x y a b c d e
f = a*x^2 + b*x + c*y;
g = d*x + e*y;

A = randn(2,5000);
B = double2sdpvar(zeros(2, 1000));

for n = 1:5000
    B(1, n) = replace(f, [x y], [A(1,n) A(2,n)]);
    B(2, n) = replace(g, [x y], [A(1,n) A(2,n)]);
end 
```

Do you have any suggestion to make it run faster?

**Comment by johanlofberg** [comment, marked as answer]
Why don't you simply create the expression directly

`a*A(1,:).^2 + b*A(1,:) + c*A(2,:)`

replace will be hard to get fast

**Reply by htadashi** [reply]
Got it! Thanks :)
